apps/src/applab/firebaseStorage.js

```diff
@@ -249,36 +254,51 @@ FirebaseStorage.deleteRecord = function (tableName, record, onComplete, onError)
  * @param {string} tableName Table to listen to.
  * @param {function (Object, RecordListener.EventType)} onRecord Callback to call when
  * a change occurs with the record object (described above) and event type.
+ * @param {function (string)} onWarning Callback to call with an warning to show to the user.
  * @param {function (string, number)} onError Callback to call with an error to show to the user and
  *   http status code.
  */
-FirebaseStorage.onRecordEvent = function (tableName, onRecord, onError) {
+FirebaseStorage.onRecordEvent = function (tableName, onRecord, onWarning, onError, includeAll) {
   if (typeof onError !== 'function') {
     throw new Error('onError is a required parameter to FirebaseStorage.onRecordEvent');
   }
   if (!tableName) {
     onError('Error listening for record events: missing required parameter "tableName"', 400);
     return;
   }
+  if (listenedTables.includes(tableName)) {
+    onWarning(`onRecordEvent was already called for table "${tableName}". To avoid 
+unexpected behavior in your program, you should only call onRecordEvent once 
+per table and use if/else statements to handle the different event types.`);
+  }
+  listenedTables.push(tableName);
+
+  getLastRecordId(tableName).then(lastId => {
+    const recordsRef = getRecordsRef(Applab.channelId, tableName);
+    // CONSIDER: Do we need to make sure a client doesn't hear about updates that it triggered?
```

Comment:
I suppose it is no long under consideration. I am of the opinion that creates from this client SHOULD be included. For example, in a chat app, you can use `onRecordEvent` (with `includeAll = true`) to update your chat window, and not care which client generated the chat message.

If this becomes confusing in classrooms, we can reconsider. It might be hard to change this in a backward-compatible manner at that point, so if you think we might not want this behavior then perhaps now is a good time to discuss.